examples/sfu/sfu/signal.rs

```diff
@@ -0,0 +1,150 @@
+use anyhow::Result; use hyper::server::conn::AddrStream;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Request, Response, Server, StatusCode};
+use std::net::SocketAddr;
+use flume::Receiver;
+use flume::Sender;
+use futures::{sink::SinkExt, stream::StreamExt};
+use hyper_tungstenite::{tungstenite, HyperWebsocket};
+use std::convert::Infallible;
+use tungstenite::Message;
+use uuid::Uuid;
+use serde::{Deserialize, Serialize};
+use tokio_util::codec::{BytesCodec, FramedRead};
+
+static INDEX: &str = "examples/sfu/index.html";
+static NOTFOUND: &[u8] = b"Not Found";
+
+fn not_found() -> Response<Body> {
+    Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .body(NOTFOUND.into())
+        .unwrap()
+}
+
+async fn simple_file_send(filename: &str) -> Result<Response<Body>, anyhow::Error> {
+    // Serve a file by asynchronously reading it by chunks using tokio-util crate.
+    if let Ok(file) = tokio::fs::File::open(filename).await {
```

Comment:
If `filename` is user controlled this can trigger a [path traversal vulnerability](https://owasp.org/www-community/attacks/Path_Traversal).